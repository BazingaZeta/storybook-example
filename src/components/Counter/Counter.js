import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/Button'

import { Wrapper, H1 } from '../typography'

const Counter = ({ onClick, label, header, count }) => (
  <Wrapper>
    <H1>{header}</H1>
    <Button onClick={onClick} primary size="large" label={label} />
    <div>
      Count: <tt>{count}</tt>
    </div>
  </Wrapper>
)

Counter.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  header: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired
}

export default Counter
