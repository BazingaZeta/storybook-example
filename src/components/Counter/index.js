import React from 'react'
import { useCountValue, useActions } from '../../features/counter'
import PureCounter from './Counter'

const Counter = () => {
  /**
   *  Get count value from Redux store. We defined selector
   *  (state => state.counter.value) inside counter feature folder,
   *  to make component global state agnostic
   */
  const count = useCountValue()

  /** Create incrementCounter action, using custom hook from feature */
  const { incrementCounter } = useActions()

  return (
    <PureCounter
      header="Counter"
      count={count}
      label="Increment by one"
      onClick={incrementCounter}
    />
  )
}

export default Counter
