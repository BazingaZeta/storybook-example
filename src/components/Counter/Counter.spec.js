import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Counter from './Counter'

describe('components > Counter > Counter', () => {
  const onClick = jest.fn()
  const header = 'some header'
  const label = 'plus me up!'
  const count = 3

  let component
  let button

  beforeAll(() => {
    component = render(<Counter onClick={onClick} header={header} label={label} count={count} />)
    button = component.getByRole('button')
    fireEvent.click(button)
  })

  it('rendered without crashing', () => {
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('passed click on to onClick', () => {
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
