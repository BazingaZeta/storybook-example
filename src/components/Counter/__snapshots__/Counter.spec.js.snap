// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Counter > Counter rendered without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-gtssRu evpqoY"
  >
    <h1
      class="sc-dlnjPT bgFMnT"
    >
      some header
    </h1>
    <button
      class="sc-bdnylx lolljv"
      type="button"
    >
      plus me up!
    </button>
    <div>
      Count: 
      <tt>
        3
      </tt>
    </div>
  </div>
</DocumentFragment>
`;
